### PR TITLE
HSV field color detector

### LINKED ISSUE #82: HSV green detector for demo purposes
As discussed in #59 the parameter gets removed but should be added with a propper implementation later on.

### PR DESCRIPTION
**Pull request by Flova**
This pull request adds an HSV field color detector, for faster configurations e.g. at demonstrations. Closes #82.

### CODE REVIEW

File: bitbots_vision/src/bitbots_vision/vision.py [hunk 1]
```diff
@@ -232,6 +232,17 @@ def _configure_vision(self, config, level):
                     config,
                     self._package_path)
 
+        # Check if params changed
+        if ros_utils.config_param_change(self._config, config,
+                r'^green_color_detector_|field_color_detector_use_hsv'):
+            # Deactivate dynamic color space
```
Comment (JanGut) [inline bitbots_vision/src/bitbots_vision/vision.py]: As this is a duplicate of the code above, I suggest outsourcing this.

File: bitbots_vision/cfg/Vision.cfg [hunk 2]
```diff
@@ -119,6 +120,14 @@ group_neural_network_type.add("ball_candidate_rating_threshold", double_t, 0, "A
...
 
+group_hsv_field_color_detector.add("field_color_detector_use_hsv", bool_t, 0, "Should the dummy green HSV detector be used instead", None)
+group_hsv_field_color_detector.add("green_color_detector_lower_values_h", int_t, 0, "Lower bound for the green color detector hue", min=0, max=180)
+group_hsv_field_color_detector.add("green_color_detector_lower_values_s", int_t, 0, "Lower bound for the green color detector saturation", min=0, max=255)
+group_hsv_field_color_detector.add("green_color_detector_lower_values_v", int_t, 0, "Lower bound for the green color detector value/brightness", min=0, max=255)
+group_hsv_field_color_detector.add("green_color_detector_upper_values_h", int_t, 0, "Upper bound for the green color detector hue", min=0, max=180)
+group_hsv_field_color_detector.add("green_color_detector_upper_values_s", int_t, 0, "Upper bound for the green color detector saturation", min=0, max=255)
+group_hsv_field_color_detector.add("green_color_detector_upper_values_v", int_t, 0, "Upper bound for the green color detector value/brightness", min=0, max=255)
```
Comment (NFiedler) [inline bitbots_vision/cfg/Vision.cfg]: why do you switch between field and green color detector?

File: bitbots_vision/src/bitbots_vision/vision.py (hunk 1 above)
Comment (Flova) [reply bitbots_vision/src/bitbots_vision/vision.py]: Which part is duplicated? The parameters in the are different in the both cases.

File: bitbots_vision/cfg/Vision.cfg (hunk 2 above)
Comment (Flova) [reply bitbots_vision/cfg/Vision.cfg]: I thought, we have the field color detector, which is normally the pixellist one. Than we have the ability to use a hsv one instead. This hsv field color detector is basically a hsv green color detector, that could be used for other stuff. See the line detector, which uses a white color detector.

Comment (NFiedler) [reply bitbots_vision/cfg/Vision.cfg]: I think it is confusing and a bit misleading as we try to approximate the green color of the field specifically. Additionally, you add green color parameters to the field color group.

**Comment by Flova** [comment]
We can rename it if you want.

**Pull request merged**